stan/math/prim/scal/fun/lower_reg_inc_gamma.hpp

```diff
@@ -0,0 +1,194 @@
+#ifndef STAN_MATH_PRIM_SCAL_FUN_LOWER_REG_INC_GAMMA_HPP
+#define STAN_MATH_PRIM_SCAL_FUN_LOWER_REG_INC_GAMMA_HPP
+
+#include <boost/math/tools/promotion.hpp>
+#include <stan/math/prim/scal/fun/lgamma.hpp>
+#include <stan/math/prim/scal/fun/gamma_p.hpp>
+#include <stan/math/prim/scal/fun/inv.hpp>
+#include <stan/math/prim/scal/fun/digamma.hpp>
+#include <stan/math/prim/scal/fun/is_nan.hpp>
+#include <stan/math/prim/scal/fun/grad_reg_inc_gamma.hpp>
+#include <stan/math/prim/scal/err/domain_error.hpp>
+#include <limits>
+
+namespace stan {
+  namespace math {
+
+    template <int IMPL = 0>
+    struct lower_reg_inc_gamma {
+      template <int ARG, typename T1, typename T2>
+      static
+      typename boost::math::tools::promote_args<T1, T2>::type
+      grad(T1 a, T2 z, double precision = 1e-10, int max_steps = 1e5) {
+        typedef typename boost::math::tools::promote_args<T1, T2>::type TP;
+        TP d = std::numeric_limits<TP>::quiet_NaN;
+        return d;
+      }
+    };
+
+
+    template <>
+    struct lower_reg_inc_gamma<1> {
```

Comment:
it was simpler when I ran tests on 8 different implementations, can you suggest something specific? I don't know how separate functions would fit with our code organization.  For example there's a dispatching function and three implementations, what do I call them?